src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -201,8 +201,8 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
     override def run(): Unit = {
       super.run()
       exitingSpecialize {
-        FunctionClass.seq.map(_.info)
-        TupleClass.seq.map(_.info)
+        FunctionClass.seq.take(MaxFunctionAritySpecialized + 1).foreach(_.info)
```

Comment:
puzzling :-)

Reply:
I understand the reason for MaxFunctionAritySpecialized thanks to the test, but why +1?

Reply:
There is a Scalafix rule now that calculates the probability of an off-by-one error and injects an adjusting `+ 1` bzw. `- 1`.